instagram/src/Authentication/Authenticate.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+
+
+const authentication = ConditionalComponent => PassedComponent => {
+    return class Authenticate extends React.Component {
+        constructor(props){
+            super(props);
+            this.state = {
+                loggedIn: false
+            };
+        }
+
+        componentDidMount(){
+            if (localStorage.getItem('username')){
+                this.setState({
+                    loggedIn: true
+                })
+            }
+        }
+        
+        render(){
+            if (this.state.loggedIn === false) {
```

Comment:
You could cut this down to only one line of code by using a ternary statement {this.state.loggedIn ? <ConditionalComponent /> :  <PassedComponent />}